src/main/scala/org/coroutines/AstCanonicalization.scala

```diff
@@ -122,6 +121,48 @@ trait AstCanonicalization[C <: Context] {
       (decls, q"$localvarname")
     case q"$selector[..$tpts](...$paramss)" if tpts.length > 0 || paramss.length > 0 =>
       // application
+
+      // We first need to see which parameters, if any, are by-name. Those that are
+      // should not be canonicalized.
+
+      // Maps to the parameter lists, saying whether or not they are by-name.
+      val byNameParams: immutable.Seq[immutable.Seq[Boolean]] = {
+        // Check that `selector` has a non-trivial symbol. If it doesn't, we assume
+        // that there were no by-name parameters.
+        if (selector.symbol != null && selector.symbol != NoSymbol) {
```

Comment:
Related to https://github.com/storm-enroute/coroutines/pull/27#discussion_r70301309. 

I think lambdas were problematic. There might have been more cases, though. I should have documented this more heavily at the time.